src/class_sqlThread.py

```diff
@@ -29,22 +18,144 @@
+
+    conn, cur = connection_build()
+
+    def __init__(self):
+        self._current_level = None
+        self.max_level = 11
+
+    def __get_current_settings_version(self):
+        """
+            Upgrade Db with respect to their versions
+        """
+        item = '''SELECT value FROM settings WHERE key='version';'''
+        parameters = ()
+        self.cur.execute(item, parameters)
+        return int(self.cur.fetchall()[0][0])
+
+    def _upgrade_one_level_sql_statement(self, file_name):
+        """
+            Execute SQL files and queries
+        """
+        try:
+            if int(file_name) == 8:
+                res = self.cur.execute('''PRAGMA table_info('inbox');''')
+                print("""""""""""""""-----------res""""""""""""""")
+                print(res)
+            print("=======================")
+            with open(os.path.join(root_path, "src/sql/init_version_{}.sql".format(file_name))) as sql_file:
+                sql_as_string = sql_file.read()
+            self.cur.executescript(sql_as_string)
+        except Exception as err:
+            if str(err) == 'table inbox already exists':
+                return "ERROR trying to upgrade database. Error message: table inbox already exists"
+            else:
+                sys.stderr.write(
+                    'ERROR trying to upgrade database. Error message: %s\n' % str(err))
+                os._exit(0)
+
+    @property
+    def sql_schema_version(self):
+        self._current_level = self.__get_current_settings_version()
+
+    @sql_schema_version.setter
+    def sql_schema_version(self, cur, conn, current_level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (current_level + 1,)
+        self.cur.execute(item, parameters)
+        self._current_level = self.__get_current_settings_version()
+
+    # @sql_schema_version.setter
+    def upgrade_to_latest(self, cur, conn, current_level=None):
+        """
+            Initialise upgrade level
+        """
+
+        # Declare variables
+        self.conn = conn
+        self.cur = cur
+        if current_level:
+            self._current_level = current_level
+        else:
+            self._current_level = self.__get_current_settings_version()
+        self.max_level = 11
+
+        # call upgrading level in loop
+        for l in range(self._current_level, self.max_level):
+            if int(l) == 1:
+                continue
+            self._upgrade_one_level_sql_statement(l)
+            # self.increment_settings_version(l)
+            self._current_level = l
+
+    def increment_settings_version(self, level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (level + 1,)
+        self.cur.execute(item, parameters)
+
+
+class sqlThread(threading.Thread, UpgradeDB):
```

Comment:
maybe an `UpgradeDB` object would be an attribute of the `sqlThread` rarther than `sqlThread` inheriting it. That way we isolate the DB stuff and can test it without threading, and the thread would need to use the same interfaces as the test.

Reply:
Changes done,
Refactored code and kept in Upgrade DB,